superset/viz.py

```diff
@@ -2050,6 +2059,10 @@ def run_extra_queries(self) -> None:
             qry["groupby"] = [col]
             metric = flt.get("metric")
             qry["metrics"] = [metric] if metric else []
+            QueryContext(
+                datasource={"id": self.datasource.id, "type": self.datasource.type},
+                queries=[qry],
+            ).raise_for_access()
```

Comment:
Is there some reason we can't just call
```python
security_manager.raise_for_access(datasource=self.datasource.id)
```
here?

Reply:
yeah, because the datasource alone isn't enough to determine if a user should be able to run the query. We also need details about the metrics and columns being queried for the filter box

Reply:
by adding queries you're just validating that all columns and metrics are not duplicates with each other.
is that what you're trying to achieve?

Reply:
we have some special logics for metric/column access in the internal security manager.

Reply:
Just so I don't misunderstand something, isn't `QueryContext.raise_for_access()` mostly just calling `SupersetSecurityManager.raise_for_access()`? I tried to look at what other assertions it does, but it's mostly just validating that there are no duplicate columns/metric if I'm not mistaken (to echo what @amitmiran137 said).

Reply:
`raise_for_access` is a public API though, and anyone with a custom SecurityManager may overwrite it. In our case, we have metric level permissions in some databases, and without this check, the query goes through to the database and fails there. Even worse, if the chart was already cached, we still load it for the user without permission because we don't call `raise_for_access` on it. does that make sense?

Reply:
Ok, now I understand, thanks for the detailed explanation. It does appear this is the simplest solution to this problem, given the fact that Filter Box is due to be deprecated soon anyway.